module/module.go

```diff
@@ -268,19 +253,21 @@ func (m *Module) GetParentResource(ctx context.Context, name resource.Name) (res
 }
 
 func (m *Module) connectParent(ctx context.Context) error {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	if m.parent == nil {
-		if err := CheckSocketOwner(m.parentAddr); err != nil {
-			return err
-		}
-		// TODO(PRODUCT-343): add session support to modules
-		rc, err := client.New(ctx, "unix://"+m.parentAddr, m.logger, client.WithDisableSessions())
-		if err != nil {
-			return err
-		}
-		m.parent = rc
+	if err := CheckSocketOwner(m.parentAddr); err != nil {
+		return err
+	}
+
+	// NOTE(benjirewis): moduleLoggers may be creating the client connection
+	// below, so use a different logger here to avoid a deadlock where the
+	// client connection tries to recursively connect to the parent.
```

Comment:
Made a decision here (see `NOTE`) due to failing tests that don't have any `parent` to connect to. Let me know what you think.

Reply:
Trying to get rid of this logic (and simply return an error in the event that connection establishment back to the parent fails) by using a mock robot server in a few different tests; please hold off on reviewing.

Reply:
Got rid of this logic; we now error in the event that connection establishment back to parent fails and I'm now using a mock robot service in a new `testutils/module_utils.go` file. Feel free to review now.